fix(addon): restore and move ie11 support to separate preset

zhzz opened the pull request:
As a continue of the #226. Inspired by [storybookjs/addon-ie11](https://github.com/storybookjs/addon-ie11).

The #187 broke the addon's ie11 support multiple times:

1. by using [`src/shared.ts`](https://github.com/creevey/blob/318ac628/src/shared.ts) that doesn't get transpiled for ie11
2. by importing [`@tesing-library`](https://github.com/creevey/creevey/blob/318ac628/src/client/addon/withCreevey.ts#L18), that distributes es6
3. by using [`fetch`](https://github.com/creevey/creevey/blob/318ac628/src/client/addon/withCreevey.ts#L208) that requires a polyfill

The separate preset will localize all the staff required for ie11 support in a single place. Also it will allow to opt the support in only for these who need it.

I've not chosen the final way to use the preset yet. I see two variants: an option for existing creevey addon and a separate ie11 addon. The second one looks more neat for me.

```js
export default {
  addons: [
    {
      name: 'creevey',
      options: {
        ie11: true,
      },
    },
  ],
```
```js
export default {
  addons: [
    'creevey',
    'creevey/preset/ie11',
  ],
```

wKich commented:
Yeah. I like the second variant too

wKich commented on src/client/addon/ie11/preset.ts:
```diff
@@ -0,0 +1,36 @@
+const nodeModulesThatNeedToBeParsedBecauseTheyExposeES6 = ['creevey', '@testing-library'];
+
+const include = new RegExp(
+  `[\\\\/]node_modules[\\\\/](${nodeModulesThatNeedToBeParsedBecauseTheyExposeES6.join('|')})`,
```
Does it handle only windows paths?

zhzz replied:
No, the final regexp looks like `/[\\/]node_modules[\\/](...)/` and handles both types of slashes. So it works for Windows and Unix paths.

wKich reviewed (dismissed):
I have only one question. But everything else are fine

zhzz commented:
The #231 introduced the `exports` field usage and made it necessary to add a separate entry point for `preset.js`, so Storybook could load it.

Also, I changed paths for the addon and the preset usage. My initial proposal for using the "creevey/addon" path faced the [browser-specific-code problem](https://github.com/creevey/creevey/pull/231) again, because we moved that code there.  So, now I propose the following solution, which combines a backward compatibility and a neat entry points structure.

```js
export default {
  addons: [
    'creevey',
    'creevey/preset/ie11',
  ]
}
```

zhzz commented:
Previous changes actually didn't fully work.

The last fixes:
- Restored transpilation of the client code in `babel.config.js`. It turned out that it's required for webpack < 5 (it uses acorn < 7 as a parser, which doesn't support `??` and `.?` operators)
- Fixed exports pathes in a way that works
- Completed `preset.ie11.ts`
